src/components/Quiz/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { USER_STATS_KEY } from "../../constants"
+
+import { CompletedQuizzes, QuizShareStats } from "../../types"
+
+import allQuizzesData, {
+  ethereumBasicsQuizzes,
+  usingEthereumQuizzes,
+} from "../../data/quizzes"
+
+export const getTotalQuizzesPoints = () =>
+  Object.values(allQuizzesData)
+    .map((quiz) => quiz.questions.length)
+    .reduce((accumulator, currentValue) => {
+      return accumulator + currentValue
+    }, 0)
+
+export const getNumberOfCompletedQuizzes = (quizzes: CompletedQuizzes) =>
+  Object.values(quizzes)
+    .map((v) => v[0])
+    .filter((v) => v).length
+
+export const getNextQuiz = (currentQuiz?: string) => {
+  const allQuizzes = [...ethereumBasicsQuizzes, ...usingEthereumQuizzes]
+  const nextQuiz = allQuizzes.find((quiz) => quiz.id === currentQuiz)
+
+  return nextQuiz ? nextQuiz.next : undefined
+}
+
+export const updateUserStats = ({
+  quizKey,
+  quizScore,
+  numberOfCorrectAnswers,
+  setUserStats,
+}) => {
+  // Read userStats from localStorage as quiz could be standalone (out of Quiz Hub page)
+  const userStats = JSON.parse(localStorage.getItem(USER_STATS_KEY)!)
+  const { score: userScore, average, completed } = userStats
+  const completedQuizzes = JSON.parse(completed)
+  // Get previous score on quiz to compare on retry, if previous score is higher then keep it
+  const lastScore = completedQuizzes[quizKey][1]
+  // Update user score, average and save to local storage
+  const newUserScore = userScore + numberOfCorrectAnswers - lastScore
+  const newAverage = [...average, quizScore]
+  const newCompleted = JSON.stringify({
+    ...completedQuizzes,
+    [quizKey!]: [
+      quizScore === 100,
+      quizScore > lastScore ? numberOfCorrectAnswers : lastScore,
+    ],
+  })
+
+  if (numberOfCorrectAnswers > lastScore) {
+    setUserStats({
+      ...userStats,
+      score: newUserScore,
+      average: newAverage,
+      completed: newCompleted,
+    })
+
+    localStorage.setItem(
+      USER_STATS_KEY,
+      JSON.stringify({
+        ...userStats,
+        score: newUserScore,
+        average: newAverage,
+        completed: newCompleted,
+      })
+    )
+  }
+}
+
+export const shareOnTwitter = ({ score, total }: QuizShareStats): void => {
+  const url = "https://ethereum.org/quizzes"
```

Comment:
I think we'll need a redirect for this. You can add it to the [`redirects.json` file](https://github.com/ethereum/ethereum-org-website/blob/dev/redirects.json)

Reply:
@pettinarip how would this redirect be like? not sure about it. We're currently using the hardcoded url for the sharing feature of quizzes cc @corwintines

Reply:
Redirect added